Foundation/NSDictionary.swift

```diff
@@ -156,13 +156,50 @@ public class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     }
     
     public func copyWithZone(zone: NSZone) -> AnyObject {
-        NSUnimplemented()
+        if self.dynamicType === NSDictionary.self {
+            // NSDictionary is immutable; just return ourself
+            return self
+        } else if self.dynamicType === NSMutableDictionary.self {
+            // Otherwise, create a new NSDictionary object
+            
+            let newDict = NSDictionary()
+            newDict._storage = _storage
```

Comment:
generally I would prefer a better isolation of control however the compiler has some issues with NSDictionary.init(dictionary: [NSObject, AnyObject])